Sources/Jetworking/Client/ResponseHandler.swift

```diff
@@ -0,0 +1,79 @@
+import Foundation
+
+enum ResponseHandler {
+    typealias WrappedCompletion<ResponseType> = (HTTPURLResponse, Data?, Decoder, @escaping Client.RequestCompletion<ResponseType>) -> () -> Void
+    enum CompletionWrapper {
+        static func void<ResponseType>(
+            currentURLResponse: HTTPURLResponse?,
+            data: Data?,
+            decoder: Decoder,
+            completion: @escaping Client.RequestCompletion<ResponseType>
+        ) -> () -> Void
+        {
+            guard ResponseType.self is Void.Type else {
+                return { completion(currentURLResponse, .failure(APIError.unexpectedError)) }
+            }
+
+            return { completion(currentURLResponse, .success(() as! ResponseType)) }
+        }
+
+        static func decodable<ResponseType: Decodable>(
+            currentURLResponse: HTTPURLResponse?,
+            data: Data?,
+            decoder: Decoder,
+            completion: @escaping Client.RequestCompletion<ResponseType>
+        ) -> () -> Void
+        {
+            guard
+                let data = data,
+                let decodedData = try? decoder.decode(ResponseType.self, from: data)
+            else {
+                return { (completion(currentURLResponse, .failure(APIError.decodingError))) }
+            }
+            // Evaluate Header fields --> urlResponse.allHeaderFields
+
+            return { (completion(currentURLResponse, .success(decodedData))) }
+        }
+    }
+
+    // TODO: Improve this function (Error handling, evaluation of header fields, status code evalutation, ...)
+    static func evaluate<ResponseType>(
```

Comment:
I suggest to rename the main function of the `ResponseHandler` to `handle`.

Reply:
This method is now private. There are two new methods introduced to reduce the parameter complexity (the`completionWrapper` is now removed from the handle methods )